util/template/template_test.go

```diff
@@ -24,60 +24,45 @@ func processTemplate(t *testing.T, tmpl SimpleValue) SimpleValue {
-			assert.Equal(t, `some\path\with\backslashes`, newTmpl.Value)
-		})
-		t.Run("EscapedNewlineInString", func(t *testing.T) {
-			tmpl := SimpleValue{Value: `{{='some\nstring\nwith\nescaped\nnewlines'}}`}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, "some\nstring\nwith\nescaped\nnewlines", newTmpl.Value)
-		})
-		t.Run("Newline", func(t *testing.T) {
-			tmpl := SimpleValue{Value: "{{=1 + \n1}}"}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, "2", newTmpl.Value)
-		})
-		t.Run("StringAsJson", func(t *testing.T) {
-			tmpl := SimpleValue{Value: "{{=toJson('test')}}"}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, `"test"`, newTmpl.Value)
-		})
-		t.Run("ObjectAsJson", func(t *testing.T) {
-			tmpl := SimpleValue{Value: "{{=toJson({test: 1})}}"}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, `{"test":1}`, newTmpl.Value)
-		})
-		t.Run("ArrayAsJson", func(t *testing.T) {
-			tmpl := SimpleValue{Value: "{{=toJson([1, '2', {an: 'object'}])}}"}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, `[1,"2",{"an":"object"}]`, newTmpl.Value)
-		})
-		t.Run("SingleQuoteAsString", func(t *testing.T) {
-			tmpl := SimpleValue{Value: `{{="'"}}`}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, `'`, newTmpl.Value)
-		})
-		t.Run("DoubleQuoteAsString", func(t *testing.T) {
-			tmpl := SimpleValue{Value: `{{='"'}}`}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, `"`, newTmpl.Value)
-		})
-		t.Run("Boolean", func(t *testing.T) {
-			tmpl := SimpleValue{Value: `{{=true == false}}`}
-			newTmpl := processTemplate(t, tmpl)
-			assert.Equal(t, "false", newTmpl.Value)
-		})
+		testCases := map[string]struct {
+			input, want string
+		}{
+			"ExprSingleQuotes":             {input: "{{='test'}}", want: "test"},
+			"ExprDoubleQuotes":             {input: `{{="test"}}`, want: "test"},
+			"ExprEscapedBackslashInString": {input: `{{='some\\path\\with\\backslashes'}}`, want: `some\path\with\backslashes`},
+			"ExprEscapedNewlineInString":   {input: `{{='some\nstring\nwith\nescaped\nnewlines'}}`, want: "some\nstring\nwith\nescaped\nnewlines"},
+			"ExprNewline":                  {input: "{{=1 + \n1}}", want: "2"},
+			"ExprStringAsJson":             {input: "{{=toJson('test')}}", want: `"test"`},
+			"ExprObjectAsJson":             {input: "{{=toJson({test: 1})}}", want: `{"test":1}`},
+			"ExprArrayAsJson":              {input: "{{=toJson([1, '2', {an: 'object'}])}}", want: `[1,"2",{"an":"object"}]`},
+			"ExprSingleQuoteAsString":      {input: `{{="'"}}`, want: `'`},
+			"ExprDoubleQuoteAsString":      {input: `{{='"'}}`, want: `"`},
+			"ExprBoolean":                  {input: `{{=true == false}}`, want: "false"},
+		}
+
+		for name, tc := range testCases {
+			t.Run(name, func(t *testing.T) {
+				tmpl := SimpleValue{Value: tc.input}
+				newTmpl := processTemplate(t, tmpl, map[string]string{})
+				assert.Equal(t, tc.want, newTmpl.Value)
+			})
+		}
+	})
```

Comment:
Just curious that we don't test expressions with a non-empty replaceMap? (not that I'm asking you to do it) But do you know why that would be?

Reply:
Idk either, but I think we should add some just in case